src/systems/elevator/utils/DoorTimer.hh

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * \author Nick Lamprianidis <nlamprian@gmail.com>
+ * \date January 2021
+ */
+
+#ifndef IGNITION_GAZEBO_SYSTEMS_DOOR_TIMER_HH_
+#define IGNITION_GAZEBO_SYSTEMS_DOOR_TIMER_HH_
+
+#include <chrono>
+#include <functional>
+#include <memory>
+
+#include <ignition/gazebo/System.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE
+{
+namespace systems
+{
+// Data forward declaration
+class DoorTimerPrivate;
+
+/// \brief Timer that's used to keep a door open. It has a configurable default
+/// wait duration that when exceeded, it calls a function to let the state
+/// machine know to transition to the next state. The timer also checks whether
+/// the doorway is blocked, in which case it keeps the door open until whatever
+/// blocks the doorway moves out of the way.
+class DoorTimer
+{
+  /// \brief Constructor
+  public: DoorTimer();
+
+  /// \brief Destructor
+  public: ~DoorTimer();
+
+  /// \brief Sets the default wait duration
+  /// \param[in] _waitDuration Duration
+  public: void SetWaitDuration(double _waitDuration);
```

Comment:
I see you using `chrono` types in some places, but `double` in others. It would be nice to use `chrono` everywhere for consistency and clarity.